### PR TITLE
[MM-54821] Convert "revoke_token_button.tsx" from Class Component to Function Component 

### PR DESCRIPTION
**Pull request by saakshiraut28**
#### Summary

This PR converts ./components/admin_console/revoke_token_button/revoke_token_button.tsx from Class Component to Function Component.

#### Ticket Link

Issue Number: #24782 
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 1]
```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
...
         e.preventDefault();
 
-        const response = await this.props.actions.revokeUserAccessToken(this.props.tokenId);
-        trackEvent('system_console', 'revoke_user_access_token');
+        try {
+            const response = await props.actions.revokeUserAccessToken(props.tokenId);
+            trackEvent('system_console', 'revoke_user_access_token');
 
-        if ('error' in response) {
-            this.props.onError(response.error.message);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:27]: why was this changed?

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 2]
```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
...
-export default class RevokeTokenButton extends React.PureComponent<RevokeTokenButtonProps> {
-    private handleClick = async (e: React.MouseEvent) => {
+const RevokeTokenButton: React.FC<RevokeTokenButtonProps> = (props) => {
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        if (error) {
+            props.onError(error);
+        }
+    }, [error, props]);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:26]: whats the use of this useEffect?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 3]
```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
...
+        >
+            <FormattedMessage
+                id='admin.revoke_token_button.delete'
+                defaultMessage='Delete'
+            />
+        </button>
+    );
+};
+
+export default RevokeTokenButton;
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:57]: This used to be a pure component, so please add here a `React.memo`.

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 4]
```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
...
+            props.onError(error);
+        }
+    }, [error, props]);
+
+    const handleClick = async (e: React.MouseEvent) => {
         e.preventDefault();
 
-        const response = await this.props.actions.revokeUserAccessToken(this.props.tokenId);
-        trackEvent('system_console', 'revoke_user_access_token');
+        try {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:31]: Why the try/catch?

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 5]
```diff
@@ -16,30 +16,42 @@ interface RevokeTokenButtonProps {
     onError: (errorMessage: string) => void;
 }
 
-export default class RevokeTokenButton extends React.PureComponent<RevokeTokenButtonProps> {
-    private handleClick = async (e: React.MouseEvent) => {
+const RevokeTokenButton: React.FC<RevokeTokenButtonProps> = (props) => {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:19]: There is no need to be explicit about the type. This can be rewritten as:
```suggestion
const RevokeTokenButton = (props: RevokeTokenButtonProps) => {
```

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks for the contribution! A few comments and questions on top of what Zubair already asked.

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx (hunk 4 above)
Comment (saakshiraut28) [reply webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:31]: I use it to attempt to revoke the user's token initially. However, I've come to realize that both the 'if' statement within the 'try' block and the 'catch' statement written outside serve the same purpose. Therefore, I will remove try/catch block.

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx (hunk 2 above)
Comment (saakshiraut28) [reply webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:26]: It ensures that the onError prop is called if the revocation fails. For instance, it would be better to use the useEffect hook to display the error in a dialog box or in the console.

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:26]: I would disagree here, i dont believe we need to re render the component just to send a prop back to parent. Please remove this effect and pass the callback directly in the handler

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by larkox** [comment]
@saakshiraut28 There seem to be still some comments not addressed, and lint issues. Would you mind taking a look?

**Comment by saakshiraut28** [comment]
> @saakshiraut28 There seem to be still some comments not addressed, and lint issues. Would you mind taking a look?

Yeah, I'll work on it.

File: webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx [hunk 6]
```diff
@@ -1,45 +1,48 @@
...
+        ) => Promise<ActionFunc | ActionResult> | ActionFunc | ActionResult;
     };
     tokenId: string;
     onError: (errorMessage: string) => void;
 }
 
-export default class RevokeTokenButton extends React.PureComponent<RevokeTokenButtonProps> {
-    private handleClick = async (e: React.MouseEvent) => {
+const RevokeTokenButton = (props: RevokeTokenButtonProps) => {
+    const [error, setError] = useState<string | null>(null);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/revoke_token_button/revoke_token_button.tsx:22]: we still need to remove this state

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by saakshiraut28** [comment]
Hey @M-ZubairAhmed, I'm a little confused on how to solve the lint errors. Could you please guide me on how to fix them? So far, I've been fixing them one by one by locating where the error occurred. Is there any formatting tool that I can use to make this process easier?

**Comment by M-ZubairAhmed** [comment]
> Hey @M-ZubairAhmed, I'm a little confused on how to solve the lint errors. Could you please guide me on how to fix them? So far, I've been fixing them one by one by locating where the error occurred. Is there any formatting tool that I can use to make this process easier?

Hey thank you for working on these fixes, yes sure you can run `npm run check` for eslint and for typecheck `npm run check-types` in the webapp folder

File: package-lock.json
Comment (larkox) [inline package-lock.json:1]: This change seems unrelated

**Review by larkox — APPROVED** [review]
Apart of one minor change and fix the lint, looks good to me.

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
@saakshiraut28 there are a few lint error you can try to run `npm run fix`

File: webapp/channels/package.json [hunk 7]
```diff
@@ -36,6 +36,7 @@
     "date-fns": "2.29.3",
     "dynamic-virtualized-list": "github:mattermost/dynamic-virtualized-list#3fe918b41de4cb08dbf43f1207bb58827b38e833",
     "emoji-regex": "10.2.1",
+    "eslint": "8.53.0",
```
Comment (M-ZubairAhmed) [inline webapp/channels/package.json:39]: please remove this unrelated change

File: package-lock.json
Comment (M-ZubairAhmed) [inline package-lock.json:1]: this as well

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/package.json
Comment (M-ZubairAhmed) [inline webapp/channels/package.json:1]: i should revert the changes in this file instead of deleting

Comment (saakshiraut28) [reply webapp/channels/package.json:1]: I tried to revert the changes but I'm getting a few errors.

Comment (M-ZubairAhmed) [reply webapp/channels/package.json:1]: you could try creating another fresh branch and copy over the changes

Comment (saakshiraut28) [reply webapp/channels/package.json:1]: Thanks for your help @M-ZubairAhmed. I have reverted the changes and ran the `npm run fix` command to resolve the eslint issues in the latest commits.

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/package.json
Comment (saakshiraut28) [reply webapp/channels/package.json:1]: @M-ZubairAhmed Would it be okay to reset all changes till commit `e08f820fb` where the file was first introduced?

Comment (M-ZubairAhmed) [reply webapp/channels/package.json:1]: yes if it helps it not bringing the change please go ahead

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
It's almost complete except for one last chance. Thank you for patiently working on it till the end.

**Comment by larkox** [comment]
@M-ZubairAhmed Friendly reminder to review this

**Review by yasserfaraazkhan — APPROVED** [review]
revoke sessions works as expected

**Review by M-ZubairAhmed — APPROVED** [review]
